.github/workflows/code_analysis.yml

```diff
@@ -51,7 +51,7 @@ jobs:
       id: diff
       run: |
         git fetch --depth=1 origin $GITHUB_BASE_REF 
-        git diff --name-only origin/$GITHUB_BASE_REF > changed_files.txt
+        git diff --diff-filter=AMR --name-only origin/$GITHUB_BASE_REF > changed_files.txt
```

Comment:
Why 'R' (Renamed) and not 'C' (Copied)?  (Related question is a file that is renamed and modifed marked as M, MR or R).  Assuming the answer to this last question is 'MR' or 'M', it sounds like we could just have `-diff-filter=AM`.